### PR TITLE
also cleanup generated templates

### PR DESCRIPTION
**Pull request by e2**
Before, clean only removes files in dist/.
After this, files copied from assets/templates/ are also deleted during 'clean'

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -56,7 +58,15 @@ gulp.task('deploy-assets', function (callback) {
   );
 });
 
+var generated_template_for = function(file) {
+  return file.replace(assetPath + '/templates/', themePath + '/');
+}
+
 gulp.task('clean', function () {
+  glob(assetPath + '/templates/**/*', function(err, files) {
```
Comment (karellm) [inline gulpfile.js]: I'd prefer an atomic structure with 2 tasks `clean-templates` and `clean-dist` both being called by the `clean` task.

Also is there a reason you are using `glob` vs `gulp` here? For consistency, I'd rather use gulp unless there is a good reason.

File: gulpfile.js [hunk 2]
```diff
@@ -56,7 +58,15 @@ gulp.task('deploy-assets', function (callback) {
   );
 });
 
+var generated_template_for = function(file) {
```
Comment (karellm) [inline gulpfile.js]: This should be `camelCase`.

**Comment by karellm** [comment]
Thanks, I added couple comments

**Comment by e2** [comment]
Done, except for the glob/del usage.

Reason: I don't know how.

gulp is based on streams, and here it's about deleting specific files from an output folder. This means, it's not about using streams, but:

1. glob
2. path name transforming
3. deleting specific files

So I don't know how to elegantly solve this with gulp. gulp-tap + gulp-clean?

Anyway, the code is improved, it's split into tasks and it works. If I can improve further, let me know how specifically. (This is pretty much the first time in my life I'm using gulp or npm or JavaScript really).

**Comment by e2** [comment]
https://github.com/gulpjs/gulp/blob/master/docs/recipes/delete-files-folder.md

BTW, I think the idea of "compiling template files" by replacing fixed strings referencing assets is flawed. Since it's PHP code anyway, it's best to just use a custom `asset_path()` method, as described [here](https://github.com/sindresorhus/gulp-rev/blob/master/integration.md#approach-3---php-reads-the-manifest-and-provides-asset-names) - (linked from `gulp-rev` Readme).

**Pull request merged**